test.disabled/ejabberd_tests/tests/auth_jwt_SUITE.erl

```diff
@@ -0,0 +1,121 @@
+    case list_to_integer(erlang:system_info(otp_release)) of
+        Release when Release < 18 ->
+            %% jwerl needs recent crypto features
+            [];
+        _ ->
+            [{group, jwt}]
+    end.
+
+groups() ->
+    [{jwt, [parallel], [auth_ok, auth_fail]}].
+
+suite() ->
+    escalus:suite().
+
+init_per_suite(Config1) ->
+    Config2 = setup_ejabberd_node(Config1),
+    escalus:init_per_suite(Config2).
+
+setup_ejabberd_node(Config0) ->
+    Config1 = ejabberd_node_utils:init(Config0),
+    ejabberd_node_utils:backup_config_file(Config1),
+
+    ejabberd_node_utils:modify_config_file(
+      [{auth_method, jwt},
+       {jwt_auth_opts, "{jwt_key, \"" ++ binary_to_list(?JWT_KEY) ++ "\"},
+                        {token_user_key, bookingNumber}"}
+      ], Config1),
+
+    ejabberd_node_utils:restart_application(ejabberd),
+    Config1.
+
+end_per_suite(Config) ->
+    escalus:end_per_suite(Config),
+    restore_ejabberd_node(Config).
+
+restore_ejabberd_node(Config) ->
+    ejabberd_node_utils:restore_config_file(Config),
+    ejabberd_node_utils:restart_application(ejabberd).
+
+init_per_group(_GroupName, Config) ->
+    Config.
+
+end_per_group(_GroupName, _Config) ->
+    ok.
+
+init_per_testcase(CaseName, Config) ->
+    escalus:init_per_testcase(CaseName, Config).
+
+end_per_testcase(CaseName, Config) ->
+    escalus:end_per_testcase(CaseName, Config).
+
+%%--------------------------------------------------------------------
+%% JSON Web Token tests
+%%--------------------------------------------------------------------
+
+auth_ok(_Config) ->
+    Password = generate_token(0),
+    {ok, _Conn, _ClientProps, _} = escalus_connection:start(auth_client_props(),
+                                                            [start_stream,
+                                                             stream_features,
+                                                             authenticate
+                                                            ]),
+    ok.
+
+auth_fail(_Config) ->
+    Password = generate_token(60),
+    ClientProps0 = [{username, ?USERNAME},
+                    {server, <<"localhost">>},
+                    {password, Password},
+                    {ssl, false}
+                   ],
+    {error, _} = escalus_connection:start(auth_client_props(),
+                                          [start_stream,
+                                           stream_features,
+                                           authenticate
+                                          ]),
+    ok.
+
+auth_client_props() ->
+    [{username, ?USERNAME},
+     {server, <<"localhost">>},
+     {password, Password},
+     {ssl, false}
+    ].
+
+generate_token(NbfDelta) ->
+    Now = p1_time_compat:system_time(seconds),
+    Data = #{bookingNumber => ?USERNAME,
+             exp => Now + 60,
+             nbf => Now + NbfDelta,
+             iat => Now},
+    Token = jwerl:sign(Data, #{alg => <<"HS256">>, key => ?JWT_KEY}),
```

Comment:
There is no need to assign it to `Token` and returning. It's enough to have just
```erlang
jwerl:sign(Data, #{alg => <<"HS256">>, key => ?JWT_KEY}).
```

As the last line in this function.